File: workshop1/Code/Java/solution/src/main/java/no/fortedigital/kafka/producer/Producer.java
package no.fortedigital.kafka.producer;

import org.apache.kafka.clients.producer.KafkaProducer;
import org.apache.kafka.clients.producer.ProducerRecord;

import java.util.Properties;

public class Producer {

    public static void main(String[] args) {
        String topic = "my-topic";
        String value = "hello world!";

        // Create producer properties
        Properties properties = new Properties();

        // set producer properties
        properties.setProperty("bootstrap.servers", "localhost:19092");
        properties.setProperty("security.protocol", "PLAINTEXT");
        properties.setProperty("key.serializer", "org.apache.kafka.common.serialization.StringSerializer");
        properties.setProperty("value.serializer", "org.apache.kafka.common.serialization.StringSerializer");

        // Create the producer
        KafkaProducer<String, String> producer = new KafkaProducer<>(properties);

        // create producer record
        ProducerRecord<String, String> producerRecord = new ProducerRecord<>(topic, value);

        // send data
        producer.send(producerRecord);

        // flush and close producer
        producer.flush();
    }
}

File: workshop1/2 - Code/TypeScript/solution/src/producer/Producer.ts
import {Kafka, KafkaConfig, ProducerRecord} from 'kafkajs';

const kafkaConfig: KafkaConfig = { brokers: ['localhost:19092'] }
const kafka = new Kafka(kafkaConfig)

const producer = kafka.producer()
producer.connect()
    .then(() => {
        console.log('Connected to Kafka')
        let value: ProducerRecord = {
            topic: 'purchases',
            messages: [
                { value: 'Hello KafkaJS user!' },
            ],
        };
        producer.send(value)
    })
